### PR TITLE
Make EditorInput compatible with antd

### LINKED ISSUE #1450: Make EditorInput compatible with antd Forms
Currently there is a lot of magic happening when dealing with EditorInput. If we made it compatible with antd Forms, we could use `getFieldDecorator` like with everything else.

https://github.com/ant-design/ant-design/blob/master/components/input/Input.tsx

### PR DESCRIPTION
**Pull request by Sekhmet**
Fixes #1450 

Changes:
- [x] Make `EditorInput` compatible with antd.
- [x] Use new editor in forms.
- [x] Improve performance.
- [x] Restore previous selection behaviour.

### CODE REVIEW

File: src/client/components/Editor/EditorInput.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import './EditorInput.less';
 
 class EditorInput extends React.Component {
   static propTypes = {
-    initialValue: PropTypes.string,
+    value: PropTypes.string, // eslint-disable-line react/require-default-props
```
Comment (Sekhmet) [inline src/client/components/Editor/EditorInput.js]: We have to disable this rule. If we set `defaultProp` on it, antd will complain about updating value without `setFieldsValue`.

**Comment by bonustrack** [comment]
Error should not fire if i dont unfocus the field, now its not possible to write a post without triggering the errors:
image

File: src/client/components/Editor/Editor.js [hunk 2]
```diff
@@ -118,29 +117,15 @@ class Editor extends React.Component {
...
-        ...form.getFieldsValue(['title', 'topics', 'reward', 'upvote']),
-      };
-
-      this.props.onUpdate({
-        ...values,
-        topics: values.topics.slice(0, 5),
-        title: values.title.slice(0, 255),
-      });
-    }, 0);
+    _.throttle(this.throttledUpdate, 200, { leading: false, trailing: true })();
```
Comment (jm90m) [inline src/client/components/Editor/Editor.js]: looks better than setTimeout 👍

**Comment by Sekhmet** [comment]
@bonustrack now it won't validate entire form on update (same as before), but also will prevent saving invalid data (they can be empty though).

**Pull request merged**